apps/src/templates/progress/ProgressBubbleSet.jsx

```diff
@@ -48,8 +48,20 @@ const ProgressBubbleSet = React.createClass({
     disabled: PropTypes.bool.isRequired
   },
 
+  getIcon() {
+    const { levels } = this.props;
+
+    for (var i = 0; i < this.props.levels.length; i++) {
+      if (levels[i].icon === null){
+        levels[i].icon = "fa-desktop";
+      }
+      levels[i].icon = levels[i].icon.substring(3);
+    }
+  },
```

Comment:
This method will end up mutating our props (i.e. it changes the value of this.props.levels[i].icon), which is something we want to avoid in React. I think a better approach would be to have getIcon take a level as input, and then output the icon string.

Then later in the file, you'd have something like `levelIcon={getIcon(level)}`